ls1intum/Artemis
# `General`: Add capability to remove course icon

### Checklist
#### General


#### Client

### Motivation and Context



[#5526](https://github.com/ls1intum/Artemis/issues/5526)

### Description


When Delete Icon Button is clicked, the course icon is set to null. When the Save Button is clicked, the Course Update API will be called. The Course Update API will set the course icon to null and delete the previous course icon file.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Course with Icon

1. Log in to Artemis
2. Navigate to Course Administration
3. Open Course Detail page
4. Click Edit Button
5. Click Delete Button under the Course Icon
6. Click Save
7. The Course Icon should be removed

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage


| Class/File | Branch | Line |
|------------|-------:|-----:|
| course-update.component.ts | 70.24% | 87% |

### Screenshots


**Desktop**
<img width="1425" alt="Screen Shot 2022-10-19 at 2 45 12 PM" src="https://user-images.githubusercontent.com/10885503/196694701-82c7e185-a653-4bec-901c-c37d5fc4762e.png">

**Mobile**
<img width="376" alt="Screen Shot 2022-10-19 at 2 45 28 PM" src="https://user-images.githubusercontent.com/10885503/196694726-005e23f1-aee2-4bb5-ae4b-0abed088afe1.png">
